code/modules/jobs/job_types/_job.dm

```diff
@@ -132,10 +132,29 @@
 					continue
 
 				if(G.slot)
-					if(H.equip_to_slot_or_del(G.spawn_item(H), G.slot))
-						to_chat(M, "<span class='notice'>Equipping you with [G.display_name]!</span>")
+					if(G.slot == ITEM_SLOT_BACK)
+						var/obj/item/storage/backpack/oldbag = H.get_item_by_slot(ITEM_SLOT_BACK)
+						var/metadata = M.client.prefs.active_character.equipped_gear[G.id]
+						var/obj/item/storage/backpack/newbag = G.spawn_item(null, metadata)
+						//This was much cleaner than trying to safely transfer contents in the correct orientation
+						oldbag.name = newbag.name
+						oldbag.desc = newbag.desc
+						if(oldbag.slowdown)
+							oldbag.name += " XL" //indicates it is dufflebag sized
+							oldbag.desc += "\nHolds as much as a duffel bag, and just as hard on your back too!"
+						oldbag.icon_state = newbag.icon_state
+						oldbag.item_state = newbag.item_state
```

Comment:
Can't we just remove the default bag and give them the new one as a full replacement? (IE, if they had a duffel bag before, they now have whatever bag they equipped and it doesn't hold extra)

Reply:
Probably, yes. 

This was the same thing Kat initially requested, and I talked them out of it because it's a lot cooler for players to have their custom bag in the size they prefer instead of forcing them into a small bag if they prefer duffle.